python_src/src/lamp_py/migrations/versions/performance_manager_prod/003_ae6c6e4b2df5_extend_service_id_view.py

```diff
@@ -0,0 +1,77 @@
+"""extend service_id_by_date_and_route
+
+Revision ID: ae6c6e4b2df5
+Revises: 1b53fd278b10
+Create Date: 2023-12-17 06:56:17.330783
+
+Details
+* upgrade -> extend service_id_by_date_and_route VIEW to generate values past current date
+* upgrade -> update canonical_stop_sequence to use row_number function instead of direct from static schedule
+
+* downgrade -> Nothing
+"""
+from alembic import op
+
+from lamp_py.migrations.versions.performance_manager_prod.sql_strings.strings_003 import (
+    view_service_id_by_date_and_route,
+)
+
+
+# revision identifiers, used by Alembic.
+revision = "ae6c6e4b2df5"
+down_revision = "1b53fd278b10"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.execute("DROP VIEW IF EXISTS service_id_by_date_and_route;")
+    op.execute(view_service_id_by_date_and_route)
+
+    op.create_index(
+        "ix_static_trips_composite_4",
+        "static_trips",
+        ["static_version_key", "service_id"],
+        unique=False,
+    )
```

Comment:
Updating the service_id_by_date_and_route VIEW with the extra join condition to static_trips resulted in a very extended run time to query, total event loop time went up to ~500 seconds.  

In testing, on staging, adding this index reduces query time from ~320 seconds down to ~3 secs.

Reply:
indexes for the win.